tests/tensor/test_slinalg.py

```diff
@@ -273,6 +274,97 @@ def test_solve_dtype(self):
 
             assert x.dtype == x_result.dtype
 
+
+class TestCholeskySolve(utt.InferShapeTester):
+    def setup_method(self):
+        self.op_class = CholeskySolve
+        self.op = CholeskySolve()
+        self.op_upper = CholeskySolve(lower=False)
+        super().setup_method()
+
+    def test_infer_shape(self):
+        rng = np.random.default_rng(utt.fetch_seed())
+        A = matrix()
+        b = matrix()
+        self._compile_and_check(
+            [A, b],  # aesara.function inputs
+            [self.op(A, b)],  # aesara.function outputs
+            # A must be square
+            [
+                np.asarray(rng.random((5, 5)), dtype=config.floatX),
+                np.asarray(rng.random((5, 1)), dtype=config.floatX),
+            ],
+            self.op_class,
+            warn=False,
+        )
+        rng = np.random.default_rng(utt.fetch_seed())
+        A = matrix()
+        b = vector()
+        self._compile_and_check(
+            [A, b],  # aesara.function inputs
+            [self.op(A, b)],  # aesara.function outputs
+            # A must be square
+            [
+                np.asarray(rng.random((5, 5)), dtype=config.floatX),
+                np.asarray(rng.random((5)), dtype=config.floatX),
+            ],
+            self.op_class,
+            warn=False,
+        )
+
+    def test_solve_correctness(self):
+        rng = np.random.default_rng(utt.fetch_seed())
+        A = matrix()
+        b = matrix()
+        y = self.op(A, b)
+        cho_solve_lower_func = aesara.function([A, b], y)
+
+        y = self.op_upper(A, b)
+        cho_solve_upper_func = aesara.function([A, b], y)
+
+        b_val = np.asarray(rng.random((5, 1)), dtype=config.floatX)
+
+        A_val = np.tril(np.asarray(rng.random((5, 5)), dtype=config.floatX))
+
+        assert np.allclose(
+            scipy.linalg.cho_solve((A_val, True), b_val),
+            cho_solve_lower_func(A_val, b_val),
+        )
+
+        A_val = np.triu(np.asarray(rng.random((5, 5)), dtype=config.floatX))
+        assert np.allclose(
+            scipy.linalg.cho_solve((A_val, False), b_val),
+            cho_solve_upper_func(A_val, b_val),
+        )
+
+    def test_solve_dtype(self):
+        dtypes = [
+            "uint8",
+            "uint16",
+            "uint32",
+            "uint64",
+            "int8",
+            "int16",
+            "int32",
+            "int64",
+            "float16",
+            "float32",
+            "float64",
+        ]
+
+        A_val = np.eye(2)
+        b_val = np.ones((2, 1))
+
+        # try all dtype combinations
+        for A_dtype, b_dtype in itertools.product(dtypes, dtypes):
+            A = matrix(dtype=A_dtype)
+            b = matrix(dtype=b_dtype)
+            x = self.op(A, b)
+            fn = function([A, b], x)
+            x_result = fn(A_val.astype(A_dtype), b_val.astype(b_dtype))
+
+            assert x.dtype == x_result.dtype
+
     def verify_solve_grad(self, m, n, assume_a, lower, rng):
```

Comment:
This test appears to be in the middle of `TestSolve`, because `TestSolve`'s `verify_solve_grad` and `test_solve_grad` methods are now in the new `TestCholeskySolve`, but they should be in `TestSolve`.